### ISSUE TITLE
Node Resource Group Lockdown

### ISSUE THREAD

**Issue by justindavies** [Azure/AKS#3395]
The ability to set the NRG to read only to stop accidental changes to the AKS data plane

**Comment by mosabami** [comment]
docs: https://learn.microsoft.com/en-us/azure/aks/cluster-configuration#node-restriction-preview

**Comment by jtencioc101** [comment]
This is not working; I am getting error: "_unrecognized arguments: –-nrg-lockdown-restriction-level ReadOnly_"

```
az version
Azure-cli    Azure-cli-core    Azure-cli-telemetry
-----------  ----------------  ---------------------
2.48.1       2.48.1            1.0.8
```
```
az extension list
Experimental    ExtensionType    Name           Path                                            Preview    Version
--------------  ---------------  -------------  ----------------------------------------------  ---------  ---------
False           whl              aks-preview    /home/julio/.azure/cliextensions/aks-preview    True       0.5.151
```
![image](https://github.com/Azure/AKS/assets/51888085/ce59e7c7-0739-470a-8fcb-71b99ca8f1fb)

**Comment by JSnow38** [comment]
This is not working for me either.

az version -o table
Azure-cli    Azure-cli-core    Azure-cli-telemetry
-----------  ----------------  ---------------------
2.51.0          2.51.0                1.1.0

az extension list -o table
Experimental    ExtensionType    Name                                 Path                                            Preview    Version
--------------  ---------------  --------------  ---------------------------------------------------  ---------  ---------
False                  whl                aks-preview    /home/jeff/.azure/cliextensions/aks-preview     True       0.5.151

az feature show --namespace "Microsoft.ContainerService" --name "NRGLockdownPreview" -o table
Name                                                                          RegistrationState
------------------------------------------------------- -------------------
Microsoft.ContainerService/NRGLockdownPreview   Registered

![image](https://github.com/Azure/AKS/assets/106977147/2891a674-a064-4f96-ab3a-cb95861ff615)

**Comment by justindavies** [comment]
There was a typo in the docs, where there was a weird hyphen, use this: --nrg-lockdown-restriction-level

PR incoming for the docs to be fixed today

**Comment by MXClyde** [comment]
@justindavies Any ETA on when this feature will be GA? And is there a way to enable it via the REST API?

**Comment by palma21** [comment]
Yes this can be enabled via the rest API. For reference all features can.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2024-06-02-preview/managedClusters.json#L4986 

This should GA on the upcoming months.

**Comment by 4c74356b41** [comment]
sorry, i might be a little thick, but how is that possible? you wont be able to configure any new external service? any azure file share\disk pvc, new node pool or update existing one, etc. so how would that work?

**Comment by MXClyde** [comment]
> This should GA on the upcoming months.
@palma21 Any updates on GA availability for this feature?

**Comment by MXClyde** [comment]
Is this still slated to become GA someday?
